docs/courses/step3-command_line_params/command_line_params.md

```diff
@@ -0,0 +1,93 @@
+# angr courses - step 3 - symbolic command line parameters and constraints on them
+
+The binary and source code for this course can be found [here](./).
+
+##### The binary for this tutorial
+The binary that we want to analyze expects one command line parameter (argv) when executed.
+If the parameter fulfiss certain criteria, a success string is printed.
+Imagine it as a login process that you want to crack.
+
+##### Background: Concept of symbolic values
+As we and angr dont know the correct password, this value is fully symbolic, meaning that it can have any value in the beginning.
+Look at the following C example.
+When analyzing the corresponding binary with angr, at state *a*, argv[1] is fully symbolic.
+Due to the if-statement the active path splits into two paths:
+* In the first path (if-condition is true, *b*), the constraint that argv[1] equals "test" is added.
+* In the second path (if-condition is false, *c*), the negation is added: argv[1] does not equal "test".
+```c
```

Comment:
Yes thats correct, but i'm checking for `strcmp(...) == 0` which is true if strcmp returns false.